storage/innobase/row/row0query.cc

```diff
@@ -0,0 +1,634 @@
+  {
+    const rec_t *sec_rec= btr_pcur_get_rec(&m_pcur);
+    rec_offs* sec_offsets= rec_get_offsets(sec_rec, sec_index, nullptr,
+                                           sec_index->n_core_fields,
+                                           ULINT_UNDEFINED, &m_heap);
+    /* Check if secondary record matches our search criteria */
+    RecordCompareAction action= callback.compare_record(search_tuple, sec_rec,
+                                                        sec_index, sec_offsets);
+    if (action == RecordCompareAction::PROCESS)
+    {
+      /* Lookup clustered record and process it */
+      if (!lookup_clustered_record(table, sec_index, clust_index,
+                                   sec_rec, callback, match_count))
+        break;
+    }
+    else if (action == RecordCompareAction::STOP)
+      break;
+    if (!btr_pcur_move_to_next(&m_pcur, &m_mtr)) break;
+  }
+  m_mtr.commit();
+  return (match_count > 0 || !search_tuple) ? DB_SUCCESS : DB_RECORD_NOT_FOUND;
+}
+
+bool QueryExecutor::lookup_clustered_record(dict_table_t *table,
+                                            dict_index_t *sec_index,
+                                            dict_index_t *clust_index,
+                                            const rec_t *sec_rec,
+                                            RecordCallback& callback,
+                                            ulint& match_count) noexcept
+{
+  /* Extract primary key from secondary index record */
+  dtuple_t *clust_tuple= row_build_row_ref(ROW_COPY_DATA, sec_index,
+                                           sec_rec, m_heap);
+  if (!clust_tuple) return false;
+  /* Now lookup the complete row using clustered index */
+  btr_pcur_t clust_pcur;
+  clust_pcur.btr_cur.page_cur.index= clust_index;
+
+  mtr_t clust_mtr{m_trx};
+  clust_mtr.start();
+  bool continue_processing= true;
+  bool record_processed= false;
+  dberr_t clust_err= btr_pcur_open(clust_tuple, PAGE_CUR_LE,
+                                   BTR_SEARCH_LEAF, &clust_pcur,
+                                   &clust_mtr);
+  if (clust_err == DB_SUCCESS && btr_pcur_is_on_user_rec(&clust_pcur))
+  {
+    const rec_t *clust_rec= btr_pcur_get_rec(&clust_pcur);
+    rec_offs* clust_offsets= rec_get_offsets(clust_rec, clust_index,
+                                             nullptr,
+                                             clust_index->n_core_fields,
+                                             ULINT_UNDEFINED, &m_heap);
+    /* Verify this is the exact record we want */
+    if (!cmp_dtuple_rec(clust_tuple, clust_rec, clust_index, clust_offsets))
+    {
+      ulint prev_match_count= match_count;
+      continue_processing= process_record_with_mvcc(
+        table, clust_index, clust_rec, clust_offsets, callback, &clust_mtr,
+        match_count);
+      record_processed= (match_count > prev_match_count);
+    }
+  }
+  clust_mtr.commit();
+  return record_processed ? continue_processing : true;
```

Comment:
I find it a form of obfuscation to use the ternary operator when any of the parameters is `true` or `false´. This is equivalent to the following:
```c++
  return !record_processed || continue_processing;
```